addons/l10n_hu_edi/models/l10n_hu_edi_transaction.py

```diff
@@ -0,0 +1,531 @@
+    # * Some states are 'active': in these states, the invoice number is reserved on NAV's side,
+    #       therefore a new transaction must not be created for the invoice.
+    # * Some states are final, which means that once the transaction has arrived into them, they can
+    #       no longer be updated.
+    # * Non-active final states are not blocking: the user can reset the invoice to draft,
+    # * Active final states are blocking: in those states, the user can no longer delete the invoice,
+    #       and must create a credit note it if he wants to annul it.
+    #
+    # STATE DIAGRAM
+    # * to_send, token_error --[upload]--> token_error, sent, send_error, send_timeout
+    # * to_send, token_error --[abort]--> unsent
+    # * sent, query_error --[query_status]--> sent, confirmed, confirmed_warning, rejected, query_error
+    # * send_timeout --[recover_timeout]--> to_send, sent, send_timeout
+    # * send_error: is a final state due to recording requirements (to retry, create a new transaction)
+    # * confirmed, confirmed_warning, rejected, unsent: are final states
+    #################################################################################################
+    _name = 'l10n_hu_edi.transaction'
+    _description = 'Hungarian Tax Authority Invoice Upload Transaction'
+    _order = 'create_date desc'
+    _rec_name = 'id'
+
+    state = fields.Selection(
+        selection=[
+            ('unsent', 'Unsent (aborted)'),
+            ('to_send', 'To Send'),
+            ('token_error', 'To Retry, could not get an authentication token'),
+            ('sent', 'Sent, waiting for response'),
+            ('send_error', 'Error when sending'),
+            ('send_timeout', 'Timeout when sending'),
+            ('confirmed', 'Confirmed'),
+            ('confirmed_warning', 'Confirmed with warnings'),
+            ('rejected', 'Rejected'),
+            ('query_error', 'To Retry, error when requesting status'),
+        ],
+        string='Status',
+        required=True,
+        copy=False,
+        default='to_send',
+        index='btree_not_null',
+    )
+    is_active = fields.Boolean(
+        string='Is Active',
+        help="Active transactions are those where the invoice number is potentially locked on NAV's side.",
+        compute='_compute_is_active',
+        search='_search_is_active',
+    )
+    credentials_id = fields.Many2one(
+        comodel_name='l10n_hu_edi.credentials',
+        string='Credentials',
+        required=True,
+        ondelete='restrict',
+        index='btree_not_null',
+    )
+    invoice_id = fields.Many2one(
+        comodel_name='account.move',
+        string='Invoice',
+        required=True,
+        ondelete='restrict',
+    )
+    operation = fields.Selection(
+        selection=[
+            ('CREATE', 'Create'),
+            ('MODIFY', 'Modify'),
+        ],
+        string='Upload invoice operation type',
+    )
+    index = fields.Integer(
```

Comment:
Do we allow sending everything at once? (or maybe just for now we can support one by one as in most edis?)

Reply:
Yes, you can batch up to 100 invoices at once